libraries/DateTime/contracts/DateTime.aes

```diff
@@ -114,13 +62,191 @@ contract DateTime =
     let timestamp = timestamp + second  
     timestamp
 
-// Private functions and internal logic
+  public function parse_timestamp(timestamp : int) : date_time = 
+    let dt : date_time = init_date_time(timestamp)
+    
+    let dt : date_time = p_get_year(dt)
+    let dt : date_time = p_get_month(dt)
+    let dt : date_time = p_get_day(dt)
+    let dt : date_time = dt{hour = rest(timestamp / 60 /60, 24)}
+    let dt : date_time = dt{minute = rest(timestamp / 60, 60)}
+    let dt : date_time = dt{second = rest(timestamp, 60)}
+    let wd : int = rest(timestamp / state.day_in_seconds + 4, 7)
+    let dt : date_time = dt{weekday = wd}
+    dt
+
+  public function get_year(timestamp : int) : int = 
+    let year : int = parse_timestamp(timestamp).year
+    year
+
+  public function diff_years(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let years : int = parse_timestamp(to_timestamp).year - parse_timestamp(from_timestamp).year
+    years
+
+  public function diff_months(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let from_dt : date_time = parse_timestamp(from_timestamp)
+    let to_dt : date_time = parse_timestamp(to_timestamp) 
+    let months : int = to_dt.year * 12 + to_dt.month - from_dt.year * 12 - from_dt.month
+    months
+
+  public function diff_days(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let days : int = (to_timestamp - from_timestamp) / state.day_in_seconds
+    days
+
+  public function diff_hours(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let hours : int = (to_timestamp - from_timestamp) / state.hour_in_seconds
+    hours
+
+  public function diff_minutes(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let minutes : int = (to_timestamp - from_timestamp) / state.minute_in_seconds
+    minutes
+
+  public function diff_seconds(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let seconds : int = to_timestamp - from_timestamp
+    seconds
+
+  public function is_valid_date(year : int, month : int, day : int) : bool = 
+    if(year > 1970 && month > 0 && month =< 12)
+      let days_in_month = get_days_in_month(month, year)
+      if(day > 0 && day =< days_in_month)
```

Comment:
the expression can be directly returned, no need for if, true, false, could also be true for the outer if

(true for all cases in this contracts)

Reply:
@HristiyanG how is this resolved?